dts: arm: silabs: Restructure Series 1 DTS content

yishai1999 opened the pull request:
As a continuation to https://github.com/zephyrproject-rtos/zephyr/pull/89946 - Restructure Silabs series 1 DTS content

yishai1999 committed:
dts: arm: silabs: Move xg1 socs to xg1 directory

Introduce subdirectory for xg1 socs.
As a porduct of this - `efm32_pg_1b.dtsi` and `efr32fg1p.dtsi` were
merged into `xg1.dtsi`.

Signed-off-by: Yishai Jaffe <yishai1999@gmail.com>

yishai1999 committed:
dts: arm: silabs: Move xg12 socs to xg12 directory

Introduce subdirectory for xg12 socs.

Signed-off-by: Yishai Jaffe <yishai1999@gmail.com>

yishai1999 committed:
dts: arm: silabs: Move efr32bg13 to xg13 directory

Introduce subdirectory for xg13 socs.

Signed-off-by: Yishai Jaffe <yishai1999@gmail.com>

Martinhoff-maker commented:
Thanks for the contribution @yishai1999 !
I've few questions:

- Why changing the style to declare the flash and ram size in the board .dtsi ? I see that Nordic is doing the same way but keeping the same style as other board in Silabs directory will be better I think (because I think both options is ok, there is no guidelines about this, or I don't find it).

Martinhoff-maker commented:
Thanks for the contribution @yishai1999!

I have a few questions:

- Why change the style to declare the flash and RAM size in the board .dtsi? I see that Nordic is doing it the same way, but keeping the same style as other boards in the Silabs directory would be better, I think (because I think both options are ok, there are no guidelines about this, or I can't find them).

- I think .pinctrl files have their place inside the board folder.

- We might wait for @asmellby for this one, but some efm32 boards are actually Series 1 boards. Here's a recap:

  MCU Series 1 consists of:
  EFM32 Jade Gecko (EFM32JG1/EFM32JG12/EFM32JG13)
  EFM32 Pearl Gecko (EFM32PG1/EFM32PG12/EFM32PG13)
  EFM32 Giant Gecko (EFM32GG11)
  Wireless SoC Series 1 consists of:
  EFR32 Blue Gecko (EFR32BG1/EFR32BG12/EFR32BG13)
  EFR32 Flex Gecko (EFR32FG1/EFR32FG12/EFR32FG13)
  EFR32 Mighty Gecko (EFR32MG1/EFR32MG12/EFR32MG13)

  I don't really know how to deal with that for the moment.

yishai1999 commented:
> Thanks for the contribution @yishai1999!
> 
> I have a few questions:
> 
> * Why change the style to declare the flash and RAM size in the board .dtsi? I see that Nordic is doing it the same way, but keeping the same style as other boards in the Silabs directory would be better, I think (because I think both options are ok, there are no guidelines about this, or I can't find them).
Seems to me like most silabs socs do it this way - definitely the more maintained series 2 socs. If you want though, I can do this in a different commit/PR.
> * I think .pinctrl files have their place inside the board folder.
I agree, I just didn't wan't to change unrelated things - Do you want me to do it anyway?
> * We might wait for @asmellby for this one, but some efm32 boards are actually Series 1 boards. Here's a recap:
>   MCU Series 1 consists of:
>   EFM32 Jade Gecko (EFM32JG1/EFM32JG12/EFM32JG13)
>   EFM32 Pearl Gecko (EFM32PG1/EFM32PG12/EFM32PG13)
>   EFM32 Giant Gecko (EFM32GG11)
>   Wireless SoC Series 1 consists of:
>   EFR32 Blue Gecko (EFR32BG1/EFR32BG12/EFR32BG13)
>   EFR32 Flex Gecko (EFR32FG1/EFR32FG12/EFR32FG13)
>   EFR32 Mighty Gecko (EFR32MG1/EFR32MG12/EFR32MG13)
>   I don't really know how to deal with that for the moment.

I will go over this and see if there is more work to do.

yishai1999 committed:
dts: arm: silabs: Move efm32gg11 to gg11 directory

Introduce subdirectory for gg11 socs.

Signed-off-by: Yishai Jaffe <yishai1999@gmail.com>

yishai1999 committed:
dts: silabs: move gpio_gecko.h defines to dt-bindings

Removed gpio_gecko.h and merged its content with gecko-pinctrl-s1.h
since that's its main usage.

Signed-off-by: Yishai Jaffe <yishai1999@gmail.com>

asmellby commented:
Nice!
Re: EFM32 -- PG1/JG1 belong in `xg1` and PG12/JG12 in `xg12`. GG11 and GG12 are separate MCUs (not related to a wireless SoC).

yishai1999 committed:
dts: arm: silabs: Move efm32gg12 to gg12 directory

Introduce subdirectory for gg12 socs.

Signed-off-by: Yishai Jaffe <yishai1999@gmail.com>

yishai1999 committed:
dts: silabs: align formatting for RAM and flash

Align the format in which the RAM and flash are set in the soc.

Signed-off-by: Yishai Jaffe <yishai1999@gmail.com>

yishai1999 committed:
boards: silabs: move pinctrl.dtsi files to board dirs

Moved pinctrl dtsi files for slwrb4250b and slwrb4255a from the soc
directory to the board directory as is done in other boards.

Signed-off-by: Yishai Jaffe <yishai1999@gmail.com>

yishai1999 commented:
> Nice! Re: EFM32 -- PG1/JG1 belong in `xg1` and PG12/JG12 in `xg12`. GG11 and GG12 are separate MCUs (not related to a wireless SoC), so they should likely get their own separate `gg11` and `gg12` directories.

All done.
